RecoParticleFlow/PFProducer/interface/PFTables.h

```diff
@@ -0,0 +1,194 @@
+#ifndef RecoParticleFlow_PFProducer_PFTables_h
+#define RecoParticleFlow_PFProducer_PFTables_h
+
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElement.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitFraction.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementSuperCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementGsfTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementBrem.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementTrack.h"
+#include "RecoParticleFlow/PFProducer/interface/TableDefinitions.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+namespace edm::soa {
+  TrackTableVertex makeTrackTableVertex(std::vector<reco::PFBlockElement*> const& targetSet);
+
+  TrackTableExtrapolation makeTrackTable(std::vector<reco::PFBlockElement*> const& objects,
+                                         reco::PFTrajectoryPoint::LayerType layerType);
+  TrackTableExtrapolation makeTrackTable(std::vector<const reco::PFBlockElementGsfTrack*> const& objects,
+                                         reco::PFTrajectoryPoint::LayerType layerType);
+  TrackTableExtrapolation makeTrackTable(std::vector<const reco::PFBlockElementBrem*> const& objects,
+                                         reco::PFTrajectoryPoint::LayerType layerType);
+  template <class RecTrackType>
+  TrackTableExtrapolation makeTrackTable(std::vector<RecTrackType> const& objects,
+                                         reco::PFTrajectoryPoint::LayerType layerType);
+
+  ConvRefTable makeConvRefTable(const std::vector<reco::ConversionRef>& convrefs);
+  ConvBremTable makeConvBremTable(const std::vector<reco::PFRecTrackRef>& convbrems);
+  BremTable makeBremTable(const std::vector<const reco::PFBlockElementBrem*>& brems);
+
+  RecHitTable makeRecHitTable(std::vector<const reco::PFRecHitFraction*> const& objects);
+  SuperClusterRecHitTable makeSuperClusterRecHitTable(std::vector<const std::pair<DetId, float>*> const& objects);
+  ClusterTable makeClusterTable(std::vector<const reco::PFBlockElementCluster*> const& objects);
+  SuperClusterTable makeSuperClusterTable(std::vector<const reco::PFBlockElementSuperCluster*> const& objects);
+  GSFTable makeGSFTable(std::vector<const reco::PFBlockElementGsfTrack*> const& objects);
+}  // namespace edm::soa
+
+//this is a transient data structure, made in order to collect the data that PFBlockAlgo needs
+//holds all input data to PFBlockAlgo computations
+
+template <class ClusterTable, class RecHitTable>
+class PFClusterTables {
+public:
+  ClusterTable cluster_table_;
+  RecHitTable rechit_table_;
```

Comment:
see 7.3 and 7.4
http://cms-sw.github.io/cms_coding_rules.html#7--design-and-coding-guidelines-1

Reply:
Hmm.. I didn't want to introduce setters and getters for all these members, I think I can make the PFClusterTables and PFTables structs by getting rid of the clear() method.